File: app/src/main/java/com/cartoon/cartoonvideos/PlayActivity.java
package com.cartoon.cartoonvideos;

import android.content.Context;
import android.content.Intent;
import android.content.pm.ActivityInfo;
import android.os.Bundle;
import android.os.Handler;
import android.util.Log;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.cartoon.cartoonvideos.adapter.FacebookAdsRecyclerviewAdapter2;
import com.cartoon.cartoonvideos.databinding.ActivityPlayBinding;
import com.cartoon.cartoonvideos.entity.Video;
import com.ddd.androidutils.DoubleClick;
import com.ddd.androidutils.DoubleClickListener;
import com.facebook.ads.Ad;
import com.facebook.ads.AdError;
import com.facebook.ads.AdSettings;
import com.facebook.ads.AdSize;
import com.facebook.ads.AdView;
import com.facebook.ads.AudienceNetworkAds;
import com.facebook.ads.InterstitialAd;
import com.facebook.ads.InterstitialAdListener;
import com.facebook.ads.NativeAdsManager;
import com.facebook.ads.NativeAdsManager.Listener;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;
import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.YouTubePlayer;
import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.listeners.AbstractYouTubePlayerListener;
import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.listeners.YouTubePlayerFullScreenListener;
import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.views.YouTubePlayerView;
import java.util.ArrayList;
import java.util.Collections;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;


public final class PlayActivity extends AppCompatActivity {
    @NotNull
    private InterstitialAd interstitialAd;
    private ActivityPlayBinding binding;
    private NativeAdsManager mNativeAdsManager;
    private ArrayList<Video> item;
    private final FullScreenHelper fullScreenHelper = new FullScreenHelper(this);

    private YouTubePlayer player;
    private float duration;




    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityPlayBinding.inflate(this.getLayoutInflater());

       setContentView(binding.getRoot());
       addYoutubePlayer();
        AudienceNetworkAds.initialize((Context)this);
      showRecycler();
       item = new ArrayList<Video>();
        AdSettings.addTestDevice("370a6a5c-76e5-40e6-9a1c-100cd45f6913");
        AdView adView = new AdView(this, getResources().getString(R.string.bannar), AdSize.BANNER_HEIGHT_50);


        binding.bannerContainer.addView(adView);
        adView.loadAd();
       mNativeAdsManager = new NativeAdsManager(this, getString(R.string.native_placement_id), 0);

        mNativeAdsManager.setListener(new Listener() {
            public void onAdsLoaded() {

                binding.recommendRecycler.setLayoutManager(new LinearLayoutManager(PlayActivity.this));

                binding.recommendRecycler.setAdapter(new FacebookAdsRecyclerviewAdapter2(PlayActivity.this,item,mNativeAdsManager));
            }

            public void onAdError(@Nullable AdError p0) {
                 }
        });


        mNativeAdsManager.loadAds();


       binding.recommendRecycler.setLayoutManager(new LinearLayoutManager(this));
       interstitialAd = new InterstitialAd(this, getResources().getString(R.string.interstital));
        InterstitialAdListener interstitialAdListener = (new InterstitialAdListener() {
            public void onInterstitialDisplayed(@NotNull Ad ad) {
            }

            public void onInterstitialDismissed(@NotNull Ad ad) {
              finish();
            }

            public void onError(@NotNull Ad ad, @NotNull AdError adError) {
            }

            public void onAdLoaded(@NotNull Ad ad) {
           }

            public void onAdClicked(@NotNull Ad ad) {
          }

            public void onLoggingImpression(@NotNull Ad ad) {
           }
        });

        interstitialAd.loadAd(interstitialAd.buildLoadAdConfig().withAdListener(interstitialAdListener).build());
       binding.textView.setText((CharSequence)this.getIntent().getStringExtra("title"));
        binding.share.setOnClickListener((OnClickListener)(new OnClickListener() {
            public final void onClick(View it) {
                try {
                    Intent shareIntent = new Intent(Intent.ACTION_SEND);
                    shareIntent.setType("text/plain");
                    shareIntent.putExtra(Intent.EXTRA_SUBJECT, "My application name");
                    String shareMessage = getIntent().getStringExtra("title");
                    shareIntent.putExtra(Intent.EXTRA_TEXT, shareMessage);
                    PlayActivity.this.startActivity(Intent.createChooser(shareIntent, (CharSequence)"choose one"));
                } catch (Exception var4) {
                }

            }
        }));
       doubleTapListener();
    }

    private final void doubleTapListener() {
        Log.i("123321", "135");

        DoubleClick doubleClick = new DoubleClick((new DoubleClickListener() {
            public void onDoubleClickEvent(@Nullable View view) {
                Log.i("123321", "137");
                    player.seekTo(duration + 10);
               binding.forwardText.setText("+10 sec");
                (new Handler()).postDelayed((Runnable)(new Runnable() {
                    public final void run() {
                        binding.forwardText.setText((CharSequence)"");
                    }
                }), 3000L);
            }

            public void onSingleClickEvent(@Nullable View view) {
                Log.i("123321", "135");

                binding.youtubePlayerView.performClick();
         binding.youtubePlayerView.getPlayerUiController().showUi(true);
            }
        }),300);

        binding.forward.setOnClickListener(doubleClick);
        DoubleClick doubleClickback = new DoubleClick((new DoubleClickListener() {
            public void onDoubleClickEvent(@Nullable View view) {
               player.seekTo(duration - 10);


                binding.backwardText.setText("-10 sec");
                (new Handler()).postDelayed((new Runnable() {
                    public final void run() {
                       binding.backwardText.setText("");
                    }
                }), 3000L);
            }

            public void onSingleClickEvent(@Nullable View view) {
                binding.youtubePlayerView.performClick();
               player.pause();


              player.play();


            }
        }), 300);
       binding.backward.setOnClickListener((OnClickListener)doubleClickback);
    }

    private final void addYoutubePlayer() {
        Log.i("123321", "22:" + this.getIntent().getStringExtra("id"));
        String id = getIntent().getStringExtra("id");

        final YouTubePlayerView youTubePlayerView = findViewById(R.id.youtube_player_view);
        getLifecycle().addObserver(youTubePlayerView);
        youTubePlayerView.addYouTubePlayerListener((new AbstractYouTubePlayerListener() {
            public void onReady(@NotNull YouTubePlayer youTubePlayer) {
                player=youTubePlayer;
                youTubePlayer.loadVideo(id, 0.0F);
                youTubePlayer.play();
            }

            public void onVideoDuration(@NotNull YouTubePlayer youTubePlayer, float duration2) {
                }

            public void onCurrentSecond(@NotNull YouTubePlayer youTubePlayer, float second) {
             duration=second;
            }
        }));
        youTubePlayerView.addFullScreenListener((new YouTubePlayerFullScreenListener() {
            public void onYouTubePlayerEnterFullScreen() {
                setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
               fullScreenHelper.enterFullScreen();
            }

            public void onYouTubePlayerExitFullScreen() {
                youTubePlayerView.getLayoutParams().height = ViewGroup.LayoutParams.WRAP_CONTENT;
                setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
               fullScreenHelper.exitFullScreen();
            }
        }));
    }

    public void onBackPressed() {

        if (binding.youtubePlayerView.isFullScreen()) {
           binding.youtubePlayerView.exitFullScreen();
        } else {

            if (interstitialAd.isAdLoaded()) {

              binding.youtubePlayerView.release();
                interstitialAd.show();
            } else {
                super.onBackPressed();
            }
        }

    }

    private final void showRecycler() {

        DatabaseReference ref = FirebaseDatabase.getInstance().getReference("videos");
        ref.addValueEventListener((ValueEventListener) (new ValueEventListener() {
            public void onCancelled(@NotNull DatabaseError p0) {

            }

            public void onDataChange(DataSnapshot snapshot) {

                if (snapshot.exists()) {
                    item.clear();
                    for (DataSnapshot dataSnapshot : snapshot.getChildren()) {
                        Video video = new Video(dataSnapshot.child("id").getValue().toString(),
                                dataSnapshot.child("title").getValue().toString(),
                                dataSnapshot.child("duration").getValue().toString(), "popular",
                                dataSnapshot.child("date").getValue().toString(),
                                dataSnapshot.child("views").getValue().toString());
                        item.add(video);


                    }
                }
                Collections.shuffle(item);

                binding.recommendRecycler.setAdapter((
                        new FacebookAdsRecyclerviewAdapter2(
                                PlayActivity.this, item, mNativeAdsManager)));
            }

        }));

    }



}

File: app/src/main/java/com/app/cartoonvideos/PlayActivity.kt
package com.app.cartoonvideos

import android.content.Intent
import android.content.pm.ActivityInfo
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.util.Log
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import com.app.cartoonvideos.adapter.FacebookAdsRecyclerviewAdapter
import com.app.cartoonvideos.databinding.ActivityPlayBinding
import com.app.cartoonvideos.entity.Video
import com.ddd.androidutils.DoubleClick
import com.ddd.androidutils.DoubleClickListener
import com.facebook.ads.*
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener
import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.YouTubePlayer
import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.listeners.AbstractYouTubePlayerListener
import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.listeners.YouTubePlayerFullScreenListener
import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.views.YouTubePlayerView
import kotlinx.coroutines.launch
import java.util.*
import kotlin.collections.ArrayList


class PlayActivity : AppCompatActivity() {
    val TAG="123321"
    private var interstitialAd: InterstitialAd? = null
    private lateinit var binding: ActivityPlayBinding
    private lateinit var mNativeAdsManager: NativeAdsManager
    var item: ArrayList<Video>? = null
    private val fullScreenHelper = FullScreenHelper(this)
    var player:YouTubePlayer?=null
    var duration:Float= 0F

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding= ActivityPlayBinding.inflate(layoutInflater)
        setContentView(binding.root)
        addYoutubePlayer()
        AudienceNetworkAds.initialize(this)
        showRecycler()
        item= ArrayList()
        AdSettings.addTestDevice("7aa3ca0d-98a7-43cb-b677-ed72c532ffbd")

        val  adView = AdView(this, resources.getString(R.string.bannar), AdSize.BANNER_HEIGHT_50)
        binding.bannerContainer.addView(adView)
        adView.loadAd()
        mNativeAdsManager =
            NativeAdsManager(this, getString(R.string.native_placement_id), 0)
        mNativeAdsManager.setListener(object : NativeAdsManager.Listener {

            override fun onAdsLoaded() {
                Log.i("123321", "42:Ad loaded")
                binding.recommendRecycler.apply {
                    layoutManager = LinearLayoutManager(this@PlayActivity)
                    adapter = FacebookAdsRecyclerviewAdapter(
                        this@PlayActivity,
                        item!!,
                        mNativeAdsManager
                    )
                }

            }

            override fun onAdError(p0: AdError?) {
                Log.i("123321", "ad error ${p0?.errorMessage}")
            }
        })
        mNativeAdsManager.loadAds()

        binding.recommendRecycler.layoutManager=LinearLayoutManager(this)






        interstitialAd = InterstitialAd(this, resources.getString(R.string.interstital))

        val interstitialAdListener: InterstitialAdListener = object : InterstitialAdListener {
            override fun onInterstitialDisplayed(ad: Ad) {
                // Interstitial ad displayed callback
                Log.e(TAG, "Interstitial ad displayed.")
            }

            override fun onInterstitialDismissed(ad: Ad) {
                // Interstitial dismissed callback
                Log.e(TAG, "Interstitial ad dismissed.")
                finish()
            }

            override fun onError(ad: Ad, adError: AdError) {
                // Ad error callback
                Log.e(TAG, "Interstitial ad failed to load: " + adError.errorMessage)
            }

            override fun onAdLoaded(ad: Ad) {
                // Interstitial ad is loaded and ready to be displayed
                Log.d(TAG, "Interstitial ad is loaded and ready to be displayed!")
                // Show the ad

            }

            override fun onAdClicked(ad: Ad) {
                // Ad clicked callback
                Log.d(TAG, "Interstitial ad clicked!")
            }

            override fun onLoggingImpression(ad: Ad) {
                // Ad impression logged callback
                Log.d(TAG, "Interstitial ad impression logged!")
            }
        }

        interstitialAd!!.loadAd(
            interstitialAd!!.buildLoadAdConfig()
                .withAdListener(interstitialAdListener)
                .build()
        )




        binding.textView.text = intent.getStringExtra("title")

        binding.share.setOnClickListener {
            try {
                val shareIntent = Intent(Intent.ACTION_SEND)
                shareIntent.type = "text/plain"
                shareIntent.putExtra(Intent.EXTRA_SUBJECT, "My application name")
                val shareMessage = intent.getStringExtra("title")
                shareIntent.putExtra(Intent.EXTRA_TEXT, shareMessage)
                startActivity(Intent.createChooser(shareIntent, "choose one"))
            } catch (e: Exception) {
                //e.toString();
            }
        }
        doubleTapListener()

    }

    private fun doubleTapListener() {




        val doubleClick = DoubleClick(object : DoubleClickListener {


            override fun onDoubleClickEvent(view: View?) {
            player?.seekTo(duration+10)
                binding.forwardText.text="+10 sec"

                Handler().postDelayed({
                    binding.forwardText.text=""

                }, 3000)


            }

            override fun onSingleClickEvent(view: View?) {
            binding.youtubePlayerView.performClick()
                binding.youtubePlayerView.getPlayerUiController().showUi(true)
            }
        })
        binding.forward.setOnClickListener(doubleClick)

        val doubleClickback = DoubleClick(object : DoubleClickListener {


            override fun onDoubleClickEvent(view: View?) {
            player?.seekTo(duration-10)
                binding.backwardText.text="-10 sec"

                Handler().postDelayed({
                    binding.forwardText.text=""

                }, 3000)


            }

            override fun onSingleClickEvent(view: View?) {
                binding.youtubePlayerView.performClick()
                player?.pause()
                player?.play()


            }
        })
        binding.backward.setOnClickListener(doubleClickback)
    }
    private fun addYoutubePlayer() {
        Log.i("123321", "22:" + intent.getStringExtra("id"))
        val id=intent.getStringExtra("id")!!
        val youTubePlayerView = findViewById<YouTubePlayerView>(R.id.youtube_player_view)

        lifecycle.addObserver(youTubePlayerView)
        youTubePlayerView.addYouTubePlayerListener(object : AbstractYouTubePlayerListener() {
            override fun onReady(youTubePlayer: YouTubePlayer) {
                player=youTubePlayer
                youTubePlayer.loadVideo(id, 0f)
                youTubePlayer.play()

            }

            override fun onVideoDuration(youTubePlayer: YouTubePlayer, duration2: Float) {

            }

            override fun onCurrentSecond(youTubePlayer: YouTubePlayer, second: Float) {
                duration=second
            }

        })
        youTubePlayerView.addFullScreenListener(object : YouTubePlayerFullScreenListener {
            override fun onYouTubePlayerEnterFullScreen() {
                requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
                fullScreenHelper.enterFullScreen()

            }

            override fun onYouTubePlayerExitFullScreen() {
                youTubePlayerView.layoutParams.height = ViewGroup.LayoutParams.WRAP_CONTENT;
                requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
                fullScreenHelper.exitFullScreen()

            }
        })
    }

    override fun onBackPressed() {
        if (binding.youtubePlayerView.isFullScreen()) {
            binding.youtubePlayerView.exitFullScreen();
        } else {
            if (interstitialAd!!.isAdLoaded){
                binding.youtubePlayerView.release()
                interstitialAd!!.show()
            }
            else
                super.onBackPressed()
        }
    }

    private fun showRecycler() {
        Handler(Looper.getMainLooper()).post {

        }


        val ref = FirebaseDatabase.getInstance().getReference("videos")
        ref.addValueEventListener(object : ValueEventListener {
            override fun onCancelled(p0: DatabaseError) {
                //To change body of created functions use File | Settings | File Templates.
            }

            override fun onDataChange(p0: DataSnapshot) {

                if (p0.exists()) {
                    item?.clear()
                    for (v in p0.children) {

                        val bal = Video(
                            id = v.child("id").value.toString(),
                            title = v.child("title").value.toString(),
                            duration = v.child("duration").value.toString(),
                            category = "popular",
                            views =  if(v.child("views").exists())v.child("views").value.toString() else "0k ",
                            date =  if(v.child("date").exists())v.child("date").value.toString() else "0 month "

                        )
                        item?.add(bal)
                        lifecycleScope.launch {
                            // myRepository.insert(bal)
                        }
                    }
                    item!!.shuffle()
                    binding.recommendRecycler.apply {
                        layoutManager = LinearLayoutManager(this@PlayActivity)
                        adapter = FacebookAdsRecyclerviewAdapter(
                            this@PlayActivity,
                            item!!,
                            mNativeAdsManager
                        )
                    }
                }

            }
        })

    }
}